queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/correction.py

```diff
@@ -155,12 +154,19 @@ def update_parties(business: Business, parties: dict, correction_filing_rec: Fil
         # The backend will have an id of type int
         if not party_info.get('officer').get('id') or \
                 (party_info.get('officer').get('id') and not isinstance(party_info.get('officer').get('id'), int)):
-            _create_party_info(business, correction_filing_rec, party_info)
+            _create_party_info(business, correction_filing_rec, party_info, is_sr_correction=is_sr_correction)
         else:
             # Update if id is present
             _update_party(party_info)
```

Comment:
I think we need to make sure we don't allow updating of existing parties when it's a SR correction.

Reply:
ok added a check before it

Reply:
I would suggest to pass only completing party in `parties: list` while calling `update_parties` instead of adding conditions in multiple places

Reply:
i think maybe we should validate it in the legal api if that's the case.  I only brought this up because Travis was testing something and all the existing business parties got deleted.

Reply:
@vysakh-menon-aot This is the payload we're sending for the correction filing that takes out the directors:

```
"parties":[{"officer":{"partyType":"person","firstName":"hello","lastName":"1234"},"roles":[{"roleType":"Completing Party","appointmentDate":"2023-07-05"}],"mailingAddress":{"streetAddress":"4343 55 Avenue Cres","addressCity":"Innisfail","addressRegion":"AB","postalCode":"T4G 1X4","addressCountry":"CA"}}]
```

Reply:
Then we don't need to check `is_sr_correction` in `update_parties` at all

Reply:
I've talked with Jia, we'll create another ticket for the parties fix - I'm not sure if this is even specific to SP Corrections?

It's going to slow us down and we need to have SR corrections completely done by tomorrow

Reply:
I think only pass completing party in parties will just works for _create_party_info, still need to check before _update_party (not update existing ones)

Reply:
https://github.com/bcgov/entity/issues/16978

Reply:
If we add validation to the legal api for SR corrections to check that only one new((no id value) party with role = "completing_party", then it should simplify things I think.  i.e. no additional check needs to be done for `_update_party`

With the assumption that the filer has code to save completing parties.

But this can be done in another ticket as @seeker25 pointed out.

Reply:
I just removed all changes relate to parties.